### DISCUSSION TITLE
Awareness of existing repo

### DISCUSSION THREAD

**Discussion by SRNissen** [friendlyanon/cmake-init#99] [Ideas]
*Possibly this was added since I last checked but it's not mentioned in the readme*

For monorepos I guess, but also specifically for me when I have a repository that has a number of associated projects across various languages, not all of which work well with cmake.

Example:

Say my employer has a Python program "Monolith".

Say it is decided, for speed and correctness reasons, to use C++ to implement some of the features - in an incremental fashion, slowly moving them out of .py files and into compiled libraries.

Say I would like to add one, two, maybe ten, C++ projects. Some of those projects should know about/use each other, some of them are single projects that just do One Thing, and they shouldn't know anything about any other projects.

I *could* do each of those C++ projects in separate repositories but that's not what we're going for here, these all get to live together in the Monolith repository.

---

I guess this might be outside the scope of cmake-init, but it sure would be nice to just point it at a folder and say "I want a C++ project *right here*, but be mindful this folder is already part of a git repository."

Automating everything from there (e.g. recursively backing out until you find the root so you can have the correct .github folder) would be a monstrous task I'm sure, but I'd be fine with any number of manual post-init tasks.

**Comment by friendlyanon** [comment]
When I try to adapt an existing project in such a way, I just use the contents of cmake-init generated projects as something to copy paste from and make decisions based on the project's pre-existing requirements. Adapting cmake-init to such a use-case would be a tall order. I intend it to be something a little smarter than just a GitHub repository template, since I have seen how other projects that attempt to do that just do everything so much worse.

If you also just view cmake-init as a guide on how to do things right, then its goal is basically reached in your case as well. For anything a little less general, there is also the wiki with examples, most of which are not the output of cmake-init + some extra, but adapted to what the goal of the example is, but uses cmake-init as the base.
